src/src/SelfServiceApiClient.js

```diff
@@ -283,7 +283,19 @@ export class SelfServiceApiClient {
     this.responseHandler(response);
 
     if (!response.ok) {
-      console.log("response was: ", response.status);
+      console.log(`for url ${url} response was: ${response.status}`);
+      return undefined;
+    }
+    return response;
+  }
+
+  async requestWithToken(url, method = "GET", payload = null) {
```

Comment:
We have  a `fetchWithToken` function as well.

They seem identical except for the name and the default parameters.
I think we need only one.